test/integration/features/profile.feature

```diff
@@ -2,104 +2,161 @@
+    When executing "minishift profile set <profilename>" fails
+    Then stderr should contain
+    """
+    A profile name must be provided. Run `minishift profile list` for a list of existing profiles.
+    """
 
-Scenario: User should be able to list default profile 'minishift'
-   Given Minishift has state "Running"
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
+  Examples: Empty profile name
+    | profilename |
+    | ''          |
+    | ' '         |
+    
+  Scenario Outline: As user, I cannot create profile with special character in profile name
+    When executing "minishift profile set <profilename>" fails
+    Then stderr should contain
     """
-    - minishift	Running		(Active)
+    Profile names must consist of alphanumeric characters only.
     """
-  Scenario: Getting default profile internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
 
-Scenario: Starting Minishift with profile foo
-    When executing "minishift start --profile foo" succeeds
-    Then Minishift should have state "Running"
+  Examples: Wrong profile names
+    | profilename |
+    | '.'         |
+    | '-'         |
+    | '#$'        |
+    | '_VM'       |
+    | ' test'     |
+    | '@test'     |
+    | '!profile'  |
+    | 'demo@1'    |
+    | '?aaa#'     |
+    | '%pro%'     |
+    | '&foo'      |
+    | '*te$st^'   |
+    | 'foo 123'   |
+    | 'foo    '   |
 
-  Scenario: Getting profile foo internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
+  Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
+    When executing "minishift profile set <profilename>" succeeds
+    Then profile <profilename> should have state "Does Not Exist"
+     And profile <profilename> should be the active profile
 
-  Scenario: User should be able to list 'foo' and 'minishift'
-      When executing "minishift profile list" succeeds
-      Then stdout should contain
-      """
-      - foo		Running		(Active)
-      - minishift	Running
-      """
+  Examples: Correct profile names
+    | profilename |
+    | Test-123    |
+    | profile_    |
+    | P1_name-    |
+    | vm_45       |
+    | 20          |
+    | random001z  |
+    | 00_XYZ-     |
+    | test_Pro-01 |
+    | foo         |
 
-  Scenario: User should be able set 'minishift' as the active profile
-      When executing "minishift profile set minishift" succeeds
-      Then stdout should contain
-      """
-      Profile 'minishift' set as active profile
-      """
+  Scenario: As user, I can switch between existing profiles
+    When executing "minishift profile set Test-123" succeeds
```

Comment:
Wrong indent as "Given" is not used.

Reply:
Given is not required here, is it ? other wise given will be overused